miner/worker.go

```diff
@@ -463,8 +463,13 @@ func (w *worker) mainLoop() {
 					txs[acc] = append(txs[acc], tx)
 				}
 				txset := types.NewTransactionsByPriceAndNonce(w.current.signer, txs)
+				tcount := w.current.tcount
 				w.commitTransactions(txset, coinbase, nil)
-				w.updateSnapshot()
+				// Only update the snapshot if any new transactons were added
+				// to the pending block
+				if tcount != w.current.tcount {
+					w.updateSnapshot()
+				}
```

Comment:
This count based approach seems a bit hacky tbh, can we extend `w.commitTransactions` to return a flag whether something was modified or not?

Reply:
actually, it already returns it if I see it correctly. other invocations are

```
		if w.commitTransactions(txs, w.coinbase, interrupt) {
			return
		}
```

Reply:
Ah no, that seems to be something else. Danm it, why don't we have docs on this thing.

Reply:
Sure... I actually started doing that, then noticed that the only code path where a transaction is added also increments this counter, so I reverted my change and used that instead. But I'll redo it the non-hacky way

Reply:
I'm actually not sure what the current returnvalue represents (would like to document that if I add another)... Do you know ?

Reply:
Hm, for some reason I only saw your first comment, when I answered